File: Day50/BreakPalindrome.java
import java.util.Scanner;

class BreakPalindrome {
    public String breakPalindrome(String palindrome) {
        int n = palindrome.length();
        if (n == 1) return "";
        char[] arr = palindrome.toCharArray();
        for (int i = 0; i < n / 2; i++) {
            if (arr[i] != 'a') {
                arr[i] = 'a';
                return new String(arr);
            }
        }
        arr[n - 1] = 'b';
        return new String(arr);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String palindrome = sc.nextLine();
        BreakPalindrome sol = new BreakPalindrome();
        System.out.println(sol.breakPalindrome(palindrome));
    }
}

File: Day50/BreakPalindrome.py
class Solution(object):
    def breakPalindrome(self, palindrome):
        """
        :type palindrome: str
        :rtype: str
        """
        n = len(palindrome)
        if n == 1:
            return ""
        arr = list(palindrome)
        for i in range(n // 2):
            if arr[i] != 'a':
                arr[i] = 'a'
                return "".join(arr)
        arr[-1] = 'b'
        return "".join(arr)

if __name__ == "__main__":
    palindrome = input().strip()
    sol = Solution()
    print(sol.breakPalindrome(palindrome))
